buffer.go

```diff
@@ -23,7 +23,21 @@ type buffer struct {
 func (s *buffer) Put(name string, value float64, tags Tags, timestamp int64, aggregations Aggregations, frequency AggregationFrequency) error {
 	// put the metric in the buffer
 	s.mu.Lock()
-	s.stdBuf = append(s.stdBuf, MetricToString(name, value, tags, timestamp, aggregations, frequency))
+	s.stdBuf = append(s.stdBuf, MetricToString(name, value, "", tags, timestamp, aggregations, frequency))
+	s.metricCount++
+
+	if !s.disableAutoFlush && s.metricCount >= s.flushSize {
+		stdBuf, aggBuf := s.drainBuffers()
+		go s.flushBuffers(stdBuf, aggBuf)
+	}
+	s.mu.Unlock()
+
+	return nil
+}
+
+func (s *buffer) User(name string, value float64, user string, tags Tags, timestamp int64, aggregations Aggregations, frequency AggregationFrequency) error {
```

Comment:
I don't think duplicating the logic of the `PutXXX` methods is the best way to do this.

In order to add this functionality to the buffer `PutXXX` methods, as well as to the (public) client `PutXXX` ones, I suggest we introduce functional options.

Something like: 

`Put(name string, value float64, tags Tags, timestamp int64, aggregations Aggregations, frequency AggregationFrequency, opts ...PutOption) error` 

In which we'd have a `WithUser` type that'd be a `PutOption`.

That approach would be backwards compatible because of the variadic arguments and would both improve the extensibility of the code, as well as avoid duplicating the logic across different methods.

Here's some references to functions options if you're not familiar with the concept:

- intro: https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis
- practical example: https://www.sohamkamani.com/golang/options-pattern/